### PR TITLE
Adding duo T5

### PR DESCRIPTION
**Pull request by wiltan-uw**
MS MARCO Passage subset run results:

top 10

2020-12-09 17:21:20 [INFO] evaluate_passage_ranker: precision@1 0.29524
2020-12-09 17:21:20 [INFO] evaluate_passage_ranker: recall@3    0.46508
2020-12-09 17:21:20 [INFO] evaluate_passage_ranker: recall@50   0.80635
2020-12-09 17:21:20 [INFO] evaluate_passage_ranker: recall@1000 0.84762
2020-12-09 17:21:20 [INFO] evaluate_passage_ranker: mrr         0.42082
2020-12-09 17:21:20 [INFO] evaluate_passage_ranker: mrr@10      0.41392

#####################
MRR @10: 0.4139229024943311
QueriesRanked: 105
#####################

top 50

2020-12-09 16:26:07 [INFO] evaluate_passage_ranker: precision@1 0.29524
2020-12-09 16:26:07 [INFO] evaluate_passage_ranker: recall@3    0.46508
2020-12-09 16:26:07 [INFO] evaluate_passage_ranker: recall@50   0.80635
2020-12-09 16:26:07 [INFO] evaluate_passage_ranker: recall@1000 0.84762
2020-12-09 16:26:07 [INFO] evaluate_passage_ranker: mrr         0.41811
2020-12-09 16:26:07 [INFO] evaluate_passage_ranker: mrr@10      0.40882

#####################
MRR @10: 0.40882086167800463
QueriesRanked: 105
#####################

### CODE REVIEW

File: pygaggle/rerank/transformer.py [hunk 1]
```diff
@@ -68,6 +73,56 @@ def rerank(self, query: Query, texts: List[Text]) -> List[Text]:
...
+            _, batch_scores = greedy_decode(self.model,
+                                            input_ids,
+                                            length=1,
+                                            attention_mask=attn_mask,
+                                            return_last_logits=True)
+
+            # 6136 and 1176 are the indexes of the tokens false and true in T5.
+            batch_scores = batch_scores[:, [6136, 1176]]
+            batch_scores = torch.nn.functional.softmax(batch_scores, dim=1)
+            batch_log_probs = batch_scores[:, 1].tolist()
```
Comment (ronakice) [inline pygaggle/rerank/transformer.py:116]: If this is softmax scores, then I'm assuming this is probabilities and not log probabilities?

File: pygaggle/run/evaluate_passage_ranker.py [hunk 2]
```diff
@@ -40,6 +42,8 @@ class PassageRankingEvaluationOptions(BaseModel):
     index_dir: Path
     method: str
     model: str
+    duo_model: str
+    top: int
```
Comment (ronakice) [inline pygaggle/run/evaluate_passage_ranker.py:46]: rename this maybe? top sounds weird, go with mono_hits? Any other suggestions?

File: pygaggle/run/evaluate_passage_ranker.py [hunk 3]
```diff
@@ -137,6 +150,13 @@ def main():
                      required=True,
                      type=str,
                      help='Path to pre-trained model or huggingface model name'),
+                 opt('--duo_model',
+                     type=str,
+                     help='Path to pre-trained model or huggingface model name'),
+                 opt('--top',
```
Comment (ronakice) [inline pygaggle/run/evaluate_passage_ranker.py:156]: again, name change here.

**Review by ronakice — CHANGES_REQUESTED** [review]
Okay minus a few things this is great stuff! Thanks for adding it :)

**Comment by ronakice** [comment]
It will be nice to run this on the entire set on Compute Canada for various `mono_hits` and also figure out how to cache mono results if available instead of running the entire pipeline multiple times, but these can be addressed in future PRs!

**Comment by ronakice** [comment]
Replicated in Colab!

File: pygaggle/rerank/transformer.py (hunk 1 above)
Comment (wiltan-uw) [reply pygaggle/rerank/transformer.py]: Yes, removed log from the name

File: pygaggle/run/evaluate_passage_ranker.py (hunk 2 above)
Comment (wiltan-uw) [reply pygaggle/run/evaluate_passage_ranker.py]: mono_hits sounds good, updated

**Review by ronakice — APPROVED** [review]
LGTM!

**Review by rodrigonogueira4 — APPROVED** [review]
LGTM!

**Comment by ronakice** [comment]
Great! Merging
